src/components/Calendar/Calendar.js

```diff
@@ -55,10 +56,44 @@ export default function CalendarView() {
             seteventError({ ...eventError, message: err.message});            });
 
     }
+    function getHoursDiff(startDate, endDate) {
+        const msInHour = 1000 * 60 * 60;
+
+        return Math.round(Math.abs(endDate - startDate) / msInHour);
+    }
+    function fetchWeeklyStudyEventData() {
+        // Get list of student's courses with studyHours per course
+        axios.get(`${process.env.REACT_APP_BASE_URL}/student/email/${user.email}`)
+            .then((courses) => {
+                for (let course of courses) {
+                  course.actualStudyHours = 0
+                }
+                setMyCourses(courses)
+            })
+            .catch((err) => {seteventError({ ...eventError, message: err.message})})
+
+        // Get list of study event the student made this week, map over student's courses and add actual study time
+        axios.get(`${process.env.REACT_APP_BASE_URL}study-events/${user.username}`)
+            .then((studyEventsThisWeek) => {
+                for (let event of studyEventsThisWeek) {
+                    for (let course of myCourses) {
+                        if (course.subject === event.subject && course.catalog === event.catalog) {
+                            const actualStudyHours = getHoursDiff(event.startDate, event.endDate)
+                            course.actualStudyHours += actualStudyHours
+                            setMyCourses(myCourses)
```

Comment:
since you setMyCourses twice here you are not adding the results of the second axios request to the setMyCourses array, you are replacing the content. The courses from your first request is ignored

Reply:
SonarCloud also doesn't like this part. Check out there comments. Do not forget to publish a screenshot of the e2e test result also.